File: C++/프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

string solution(string my_string, vector<vector<int>> queries) {
    string tmp_string;
    for(vector<int> query : queries) {
        tmp_string = my_string.substr(query[0], query[1] - query[0] + 1);
        reverse(tmp_string.begin(), tmp_string.end());
        my_string.replace(query[0], query[1] - query[0] + 1, tmp_string);
    }
    return my_string;
}

File: Python3/프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    for query in queries:
        # print("my_string 1:", my_string[:query[0]])
        # print("my_string 2:", ''.join(reversed(my_string[query[0]:query[1] + 1])))
        # print("my_string 3:", my_string[query[1]:])
        my_string = my_string[:query[0]] + ''.join(reversed(my_string[query[0]:query[1] + 1])) + my_string[query[1] + 1:]
    return my_string